cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/ConfigurationGroupsCollection/index.tsx

```diff
@@ -0,0 +1,191 @@
+import uuidV4 from 'uuid/v4';
+
+const styles = (): StyleRules => {
+  return {
+    eachGroup: {
+      display: 'grid',
+      gridTemplateColumns: '5fr 1fr',
+    },
+    groupContent: {
+      display: 'block',
+      padding: '0px 0',
+      width: 'calc(100%)',
+    },
+  };
+};
+
+const ConfigurationGroupsCollectionView: React.FC<ICreateContext & WithStyles<typeof styles>> = ({
+  classes,
+  configurationGroups,
+  setConfigurationGroups,
+  groupToInfo,
+  setGroupToInfo,
+}) => {
+  const [localConfigurationGroups, setLocalConfigurationGroups] = React.useState(
+    configurationGroups
+  );
+  const [activeGroupIndex, setActiveGroupIndex] = React.useState(null);
+  const [localGroupToInfo, setLocalGroupToInfo] = React.useState(groupToInfo);
+
+  function addConfigurationGroup(index: number) {
+    const newGroupID = 'ConfigGroup_' + uuidV4();
+
+    // add a new group's ID at the specified index
+    const newGroups = [...localConfigurationGroups];
+    if (newGroups.length === 0) {
+      newGroups.splice(0, 0, newGroupID);
+    } else {
+      newGroups.splice(index + 1, 0, newGroupID);
+    }
+    setLocalConfigurationGroups(newGroups);
+
+    // set the activeGroupIndex to the new group's index
+    if (newGroups.length <= 1) {
+      setActiveGroupIndex(0);
+    } else {
+      setActiveGroupIndex(index + 1);
+    }
+
+    // set the default label and description of the group
+    setLocalGroupToInfo({
+      ...localGroupToInfo,
+      [newGroupID]: {
+        label: '',
+        description: '',
+      } as IConfigurationGroupInfo,
+    });
+  }
+
+  function deleteConfigurationGroup(index: number) {
+    setActiveGroupIndex(null);
+
+    // delete a group at the specified index
+    const newGroups = [...localConfigurationGroups];
+    const groupToDelete = newGroups[index];
+    newGroups.splice(index, 1);
+    setLocalConfigurationGroups(newGroups);
+
+    // delete the corresponding information of the group
+    const { [groupToDelete]: info, ...restGroupToInfo } = localGroupToInfo;
+    setLocalGroupToInfo(restGroupToInfo);
+  }
+
+  const switchEditConfigurationGroup = (index) => (event, newExpanded) => {
+    if (newExpanded) {
+      setActiveGroupIndex(index);
+    } else {
+      setActiveGroupIndex(null);
+    }
+  };
+
+  function saveAllResults() {
+    setConfigurationGroups(localConfigurationGroups);
+    setGroupToInfo(localGroupToInfo);
+  }
+
+  return (
+    <div>
+      <Heading type={HeadingTypes.h3} label="Configuration Groups" />
+      <br />
+      <If condition={localConfigurationGroups.length === 0}>
+        <Button variant="contained" color="primary" onClick={() => addConfigurationGroup(0)}>
+          Add Configuration Group
+        </Button>
+      </If>
+
+      {localConfigurationGroups.map((groupID, i) => {
+        const configurationGroupExpanded = activeGroupIndex === i;
+        const group = localGroupToInfo[groupID];
+        return (
+          <div className={classes.eachGroup}>
```

Comment:
nit: Will probably need a key

Reply:
changed! This fixed warnings in console.